backoff.go

```diff
@@ -0,0 +1,132 @@
+package linger
+
+import (
+	"context"
+	"math"
+	"sync/atomic"
+	"time"
+)
+
+// BackoffStrategy is a function for computing delays between attempts to
+// perform some application-defined operation.
+//
+// n is the number of successive failures since the last success, including this
+// one.
+//
+// err is the error describing the operation's failure, if known. A nil error
+// does not indicate a success.
+type BackoffStrategy func(n int, err error) time.Duration
+
+// DefaultBackoffStrategy is the default strategy used by Backoff.
+//
+// It is a conservative policy favouring large delay times under the assumption
+// that the operation is expensive.
+var DefaultBackoffStrategy BackoffStrategy = ExponentialBackoff(3 * time.Second)
+
+// Backoff introduces delays between atempts to perform some application-defined
+// operation.
+type Backoff struct {
```

Comment:
Judging by the content of this struct there is no logic for Max Retry Time or something similar, rights, retries are indefinite, until context on `Backoff.Sleep()` method is cancelled.

Reply:
Yeah, since this doesn't actually execute the "operation", any "global timeout" would just be implemented using regular context features.

A lot of the Linger module is concerned with making timeouts/sleeping/etc work easily with deadlines expressed in contexts.